### DISCUSSION TITLE
Как добавить вариант доставки наложенным платежом

### DISCUSSION THREAD

**Discussion by Xprinter-dv** [cdek-it/wordpress#37] [Ideas]
На сайте планируется два варианта доставки "Посылка склад-склад (136)" одна из них предоплата на сайте, другая наложенный платёж.
Подскажите как настроить чтобы отображалось несколько вариантов доставки данного типа, доставка наложенным платежом должна быть +4% стоимости товара и доставки.

В настройках плагина есть что-то про наложенный платёж, но не понятно как это работает.

**Comment by Xprinter-dv** [comment]
Удалось разобраться с наложенным платежом, как он работает с плагином. Только остался момент, что его можно включить для любого метода "доставка СДЭК", а хотелось бы сделать его доступным только для метода склад-склад и отключить для склад-дверь

**Reply by vermorag** [reply]
Добрый день. Данного функционала в настоящий момент не существует в плагине.

**Reply by Dustor** [reply]
Такое можно решить костылями. Надо написать хук для метода  'woocommerce_available_payment_gateways'. Например, у меня на сайте вот такой код отключает возможность предоплаты для СДЭК:
	if ( isset( $available_gateways[ 'cheque' ] ) 
    	&& (strncmp($current_shipping_method, 'official_cdek', 13)==0) )
	{
		unset( $available_gateways[ 'cheque' ] );
	}
 
	return $available_gateways;
